x/delayedack/hooks.go

```diff
@@ -32,24 +32,37 @@ func (im IBCMiddleware) FinalizeRollappPackets(ctx sdk.Context, rollappID string
 	// Get the packets for the rollapp until height
 	logger.Debug("Finalizing IBC rollapp packets", "rollappID", rollappID, "state end height", stateEndHeight, "num packets", len(rollappPendingPackets))
 	for _, rollappPacket := range rollappPendingPackets {
-		logger.Debug("Finalizing IBC rollapp packet", "rollappID", rollappID, "sequence", rollappPacket.Packet.GetSequence(), "destination channel", rollappPacket.Packet.GetDestChannel())
+		logger.Debug("Finalizing IBC rollapp packet", "rollappID", rollappID, "sequence", rollappPacket.Packet.GetSequence(), "destination channel", rollappPacket.Packet.GetDestChannel(), "type", rollappPacket.Type)
 		// Update the packet status
 		im.keeper.UpdateRollappPacketStatus(ctx, rollappID, rollappPacket, types.RollappPacket_ACCEPTED)
```

Comment:
I think it's better to move it after the switch case, and set according to the error received
```
// Update the packet status
if err != nil {
	im.keeper.UpdateRollappPacketStatus(ctx, rollappID, rollappPacket, types.RollappPacket_REJECTED)
} else {
	im.keeper.UpdateRollappPacketStatus(ctx, rollappID, rollappPacket, types.RollappPacket_ACCEPTED)
}
```

Reply:
yeah but do u want to mark it as ACCEPTED?
In normal case, if the app would return err, the relayer would retry

in our case the packet will be marked as accepted and forgotten (which might be fine if by design, but probably need to be marked somehow)

Reply:
IMO it should be marked as FINALIZED - as it is finalized.
as I previously wrote I think ACCEPTED is confusing term here.
so as I see it, the packet is finalized, but it could have errors.

as for the relayer - if we're talking about application layer error which are not part of out-of-sync problem (e.g LC headers didn't yet arrive - which is not the case in our example) - by default the relayer will not retry (i.e if you simply run `rly tx transfer` command). It's true that if you restart it will try to get all non-relayed packets but if the packet itself is maleformed or it's data invalid - retry won't help unless the relayer code or the blockchain state changes.

My point is that as I see it mostly application layer errors, specifically for ics-20 which are mostly around malformed data or lack of balances wouldn't change with relayer retry (which eventually get exhausted and the packet transfer would fail) so I think it's better to say: "Hi, this packet is finalized, but we couldn't apply it as it was malformed/not enough balance".

re forgotten - I think our UI should reflect any errors in the packet by looking at the error field.